### ISSUE TITLE
wav2vec2 encoder does not disable masking during inference

### ISSUE THREAD

**Issue by kehanlu** [espnet/espnet#4768]
**Describe the bug**
In ESPNET2, wav2vec2 encoder calls fairseq `Wav2Vec2Model`. However, the `mask` parameter defaults to `True` in fairseq; this will result in always applying masking even during inference.
https://github.com/espnet/espnet/blob/d0c12f994042b7bc621f6f996f4e6bf0200232c3/espnet2/asr/encoder/wav2vec2_encoder.py#L123-L127

`fairseq/fairseq/models/wav2vec/wav2vec2.py`
https://github.com/facebookresearch/fairseq/blob/313ff0581561c7725ea9430321d6af2901573dfb/fairseq/models/wav2vec/wav2vec2.py#L515-L525
```python
def forward(
        self,
        source,
        padding_mask=None,
        mask=True,
        features_only=False,
        layer=None,
        mask_indices=None,
        mask_channel_indices=None,
        padding_count=None,
    ):
```
https://github.com/facebookresearch/fairseq/blob/313ff05815/fairseq/models/wav2vec/wav2vec2.py#L582-L600
```python
        if mask:
            x, mask_indices = self.apply_mask(
                features,
                padding_mask,
                mask_indices=mask_indices,
                mask_channel_indices=mask_channel_indices,
            )
            if not is_xla_tensor(x) and mask_indices is not None:
                # tpu-comment: reducing the size in a dynamic way causes
                # too many recompilations on xla.
                y = unmasked_features[mask_indices].view(
                    unmasked_features.size(0), -1, unmasked_features.size(-1)
                )
            else:
                y = unmasked_features
        else:
            x = features
            y = unmasked_features
            mask_indices = None
```

To fixed the bug, just pass `mask` to the forward call:
```python
 enc_outputs = self.encoders( 
     xs_pad, 
     masks, 
     mask=self.training, 
     features_only=True, 
 ) 
```

See also `fairseq/fairseq/models/wav2vec/wav2vec2_asr.py` : https://github.com/facebookresearch/fairseq/blob/313ff0581561c7725ea9430321d6af2901573dfb/fairseq/models/wav2vec/wav2vec2_asr.py#L389-L406

**Comment by sw005320** [comment]
Thanks for the report!
This sounds critical...
@simpleoier, @DanBerrebbi, could you check this?

**Comment by DanBerrebbi** [comment]
Hi, thank you for the comment, indeed we should fix this.

**Comment by kehanlu** [comment]
Thank you all
